crates/iceberg-catalog/src/service/storage/s3.rs

```diff
@@ -295,20 +370,16 @@ impl S3Profile {
         _: WarehouseIdent,
         _: TableIdentUuid,
         _: NamespaceIdentUuid,
-        data_access: &DataAccess,
-        _: Option<&S3Credential>,
-    ) -> Result<HashMap<String, String>, TableConfigError> {
-        let DataAccess {
+        DataAccess {
             vended_credentials,
             remote_signing,
-        } = data_access;
+        }: &DataAccess,
+        table_location: &str,
+        cred: Option<&S3Credential>,
+    ) -> Result<HashMap<String, String>, TableConfigError> {
         // If vended_credentials is False and remote_signing is False,
         // use remote_signing.
-        let mut remote_signing = if !vended_credentials && !remote_signing {
-            true
-        } else {
-            *remote_signing
-        };
+        let remote_signing = !vended_credentials || *remote_signing;
```

Comment:
I think once we implement STS session handling with point-in-time logout, we should default to vended_credentials as it requires less roundtrips for datasources & clients that support it.